src/App.js

```diff
@@ -0,0 +1,138 @@
+import './App.css';
+import React, { Component } from 'react';
+import { nanoid } from 'nanoid';
+import { logicalOperators, studentsInfo, comparisonOperators } from './utils';
+import Rule from './Rule';
+
+class App extends Component {
```

Comment:
I would like to see the `App.js` component only render so remove all that logic you have put and place it where it belongs, let the App component only render the say `Rule` component. Your App component is doing so much, let's try to seperate concerns